### PR TITLE
Fix/tbrom updates

### PR DESCRIPTION
**Pull request by chrpetre**
update latest example documentation and code
update generate_snapshot_batch method with new optional argument
add code to store original state of the TwinModel and retrieve it once snapshot generation in batch is done

### CODE REVIEW

File: src/ansys/pytwin/evaluate/twin_model.py [hunk 1]
```diff
@@ -1549,6 +1554,9 @@ def generate_snapshot_batch(self, batch_results: pd.DataFrame, rom_name: str, na
             columns = batch_results.columns[1::]
             outpath = []
```
Comment (EDCarman) [inline src/ansys/pytwin/evaluate/twin_model.py:1555]: Now that we're outputting either paths, or arrays, I'd suggest renaming 'outpath' to something like 'snapshots' or 'outputs' to remove ambiguity

File: src/ansys/pytwin/evaluate/twin_model.py [hunk 2]
```diff
@@ -1504,10 +1504,13 @@ def generate_snapshot(self, rom_name: str, on_disk: bool = True, named_selection
...
-    def generate_snapshot_batch(self, batch_results: pd.DataFrame, rom_name: str, named_selection: str = None):
+    def generate_snapshot_batch(
+        self, batch_results: pd.DataFrame, rom_name: str, on_disk: bool = True, named_selection: str = None
+    ):
         """
-        Generate several field snapshots based on historical batch results of the Twin, for the full field or a specific
-        named selection. It returns a list of the paths of the different snapshots written on disk.
+        Generate several field snapshots based on historical batch results of the Twin, either in memory or on disk, for
+        the full field or a specific named selection. It returns a list of the field data as an array if in memory, or a
+        list of the paths of the different snapshots written on disk.
```
Comment (EDCarman) [inline src/ansys/pytwin/evaluate/twin_model.py:1513]: Possibly add a Returns section to make this clearer e.g.
```
Returns
--------
list[str] | list[np.ndarray]
    list of paths to snapshots written to disk if on_disk == True, else list of arrays of snapshot field data
```

**Review by EDCarman — CHANGES_REQUESTED** [review]

File: src/ansys/pytwin/evaluate/twin_model.py (hunk 1 above)
Comment (chrpetre) [reply src/ansys/pytwin/evaluate/twin_model.py]: I have included these changes, thanks

File: src/ansys/pytwin/evaluate/twin_model.py (hunk 2 above)
Comment (chrpetre) [reply src/ansys/pytwin/evaluate/twin_model.py:1518]: I have included these changes and adopted them for other similar methods, thanks

**Review by EDCarman — APPROVED** [review]

**Pull request merged**